File: uploadservice/src/main/java/net/gotev/uploadservice/http/impl/HurlStack.java
package net.gotev.uploadservice.http.impl;

import net.gotev.uploadservice.http.HttpConnection;
import net.gotev.uploadservice.http.HttpStack;

import java.io.IOException;

/**
 * HttpUrlConnection stack implementation.
 * @author gotev (Aleksandar Gotev)
 */
public class HurlStack implements HttpStack {

    private boolean mFollowRedirects;
    private boolean mUseCaches;
    private int mConnectTimeout;
    private int mReadTimeout;

    public HurlStack() {
        mFollowRedirects = true;
        mUseCaches = false;
        mConnectTimeout = 15000;
        mReadTimeout = 30000;
    }

    public HurlStack(boolean followRedirects,
                     boolean useCaches,
                     int connectTimeout,
                     int readTimeout) {
        mFollowRedirects = followRedirects;
        mUseCaches = useCaches;
        mConnectTimeout = connectTimeout;
        mReadTimeout = readTimeout;
    }

    @Override
    public HttpConnection createNewConnection(String method, String url) throws IOException {
        return new HurlStackConnection(method, url, mFollowRedirects, mUseCaches,
                                       mConnectTimeout, mReadTimeout);
    }

}

File: uploadservice/src/main/java/net/gotev/uploadservice/http/hurl/HurlStack.kt
package net.gotev.uploadservice.http.hurl

import net.gotev.uploadservice.http.HttpConnection
import net.gotev.uploadservice.http.HttpStack

import java.io.IOException

/**
 * HttpUrlConnection stack implementation.
 * @author gotev (Aleksandar Gotev)
 */
class HurlStack(private val followRedirects: Boolean = true,
                private val useCaches: Boolean = false,
                private val connectTimeout: Int = 15000,
                private val readTimeout: Int = 30000) : HttpStack {

    @Throws(IOException::class)
    override fun createNewConnection(method: String, url: String): HttpConnection {
        return HurlStackConnection(method, url, followRedirects, useCaches,
                connectTimeout, readTimeout)
    }

}
